openmeter/customer/adapter/customer.go

```diff
@@ -284,6 +301,12 @@ func (a *adapter) DeleteCustomer(ctx context.Context, input customer.DeleteCusto
 					return nil, fmt.Errorf("failed to delete customer subjects: %w", err)
 				}
 
+				// Publish the customer deleted event
+				customerDeletedEvent := customer.NewCustomerDeleteEvent(*previousCustomer, deletedAt)
+				if err := repo.publisher.Publish(ctx, customerDeletedEvent); err != nil {
+					return nil, fmt.Errorf("failed to publish customer deleted event: %w", err)
+				}
+
```

Comment:
Please publish in the service layer if possible. The adapter layer is for db work.